securebanking-openbanking-uk-rcs-api/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/api/dto/consent/details/ConsentDetails.java

```diff
@@ -58,6 +60,9 @@ public abstract class ConsentDetails {
     private String serviceProviderName; // (aisp, aspsp)
     private List<FRAccountWithBalance> accounts;
 
+    @JsonIgnore
+    public abstract FRAccountIdentifier getDebtorAccount();
```

Comment:
Does this need to be on the ConsentDetails base class?

The problem with this is that for several non-payment based consents, the implementation is to return null. This is a code smell of violating the Interface Segregation Principle (the I in SOLID).

Perhaps you could create a base PaymentsConsentDetails which extends ConsentDetails and includes this method. Then all of the Payments can extend from this.

Reply:
Done!